### DISCUSSION TITLE
Programmatically generate access token, for alternative auth flow

### DISCUSSION THREAD

**Discussion by usaadi** [fastapi-users/fastapi-users#1136] [Q&A] [answered]
Hello...

I'm trying to implement google social login in a mobile app that is using fastapi-users as the backend.  I can retrieve the "idToken" from google, and now I need to verify it on the backend and retrieve an access token and send it back to the app.

To verify the token, I can follow the instructions on the page: https://firebase.google.com/docs/auth/admin/verify-id-tokens#python

But I haven't found the way to generate an access token for this use case.  Is this possible and if so how can it be done?

Thanks in advance

**Comment by frankie567** [comment, marked as answer]
If you performed the full authentication process in your mobile app, FastAPI Users won't be able to anything with the token you have, they are tied to a Google identity.

I think your best bet is to use libraries for this specific use case, like [FastAPI Cloud Auth](https://github.com/tokusumi/fastapi-cloudauth) which are able to validate the token on the server side. Doing this, you don't really need FastAPI Users.

**Reply by usaadi** [reply]
I was thinking that I could retrieve the email after decoding the id token that originated from Google. But I'm not sure if I could do that. In any case, since this is not the recommended way, I will be looking to other libraries for this app. Thanks for the pointer.
